Modify release notes (#4865)
openservicemesh/osm: docs/release_notes.md

### Notable changes

- OSM certificate provider is now configured using the new CRD, MeshRootCertificate

>   - Custom trust domains (i.e. certificate CommonNames) are now supported

Reviewer: @steeling Just want to confirm this is accurate

Reviewer 2: I thought we are no longer using the CRD?

Reviewer: See https://github.com/openservicemesh/osm/pull/4865#discussion_r914183380

> - The authentication token used to configure the Hashicorp Vault certificate provider can now be passed in using a secretRef

Reviewer: Thanks for making these updates!

Author: Agreed, waiting on @steeling to confirm

Reviewer 2: the custom trust domain is only plumbed through on MRC's. it would take a small amount of work to add the flag to the binary and pass to the cert manager. have we cut a release yet? if not i can do for tomorrow

Author: Yeah no release has been cut yet right @nojnhuh

Reviewer 3: Correct, I was going to make sure this was merged before cutting the release.

Reviewer 2: done now with https://github.com/openservicemesh/osm/pull/4877

- Along with root certificate rotation we support custom trust domains, as well as rotating to new trust domains with no downtime.
- Envoy has been updated to v1.22 and uses the `envoyproxy/envoy-distroless` image instead of the deprecated `envoyproxy/envoy-alpine` image.
  - This means that `kubectl exec -c envoy ... -- sh` will no longer work for the Envoy sidecar
- Added support for Kubernetes 1.23 and 1.24
- `Rate limiting`: Added capability to perform local per-instance [rate limiting of TCP connections and HTTP requests](https://release-v1-2.docs.openservicemesh.io/docs/guides/traffic_management/rate_limiting).

> - Statefulsets and headless services should now work as expected

Reviewer 4: ```suggestion
- Statefulsets and headless services have been fixed and work as expected
```

- Root certificate rotation is now supported through the new CRD, MeshRootCertificate.
- Along with root certificate rotation we support custom trust domains, as well as rotating to new trust domains with no downtime.
- Envoy has been updated to v1.22 and uses the `envoyproxy/envoy-distroless` image instead of the deprecated `envoyproxy/envoy-alpine` image.
  - This means that `kubectl exec -c envoy ... -- sh` will no longer work for the Envoy sidecar
- Added support for Kubernetes 1.23 and 1.24

> - Removed support for Kubernetes 1.21

Reviewer 3: I'd remove this since it's already mentioned in the `Breaking Changes` section.
